src/main/java/org/cryptacular/util/CertUtil.java

```diff
@@ -454,4 +479,146 @@ public static String authorityKeyId(final X509Certificate cert) throws EncodingE
   {
     return CodecUtil.hex(new ExtensionReader(cert).readAuthorityKeyIdentifier().getKeyIdentifier(), true);
   }
+
+
+  /**
+   * PEM encodes the given certificate with the provided encoding type.
+   *
+   * @param certificate X.509 certificate.
+   * @param encodeType Type of encoding. {@link EncodeType#X509} or {@link EncodeType#PKCS7}
+   *
+   * @return PEM-encoded certificate headed and footer defined by {@link EncodeType} and data wrapped at 64
+   * characters per line.
+   */
+  public static String encodeCert(final X509Certificate certificate, final EncodeType encodeType)
+  {
+    try {
+      return encodeCert(certificate.getEncoded(), encodeType);
+    } catch (CertificateEncodingException e) {
+      throw new RuntimeException("Error getting encoded X.509 certificate data", e);
+    }
+  }
+
+  /**
+   * Converts the given der encoding to PEM encoding.
+   *
+   * @param der DER encoding of certificate with specified encoding.
+   * @param encodeType Encoding type that define the header and footer.
+   *
+   * @return PEM-encoded certificate headed and footer defined by {@link EncodeType} and data wrapped at 64
+   * characters per line.
+   */
+  public static String encodeCert(final byte[] der, final EncodeType encodeType)
+  {
+    final Base64Encoder encoder = new Base64Encoder(64);
+    final ByteBuffer input = ByteBuffer.wrap(der);
+    final CharBuffer output = CharBuffer.allocate(encoder.outputSize(der.length) + 100);
+    output.append("-----BEGIN ").append(encodeType.getType()).append("-----");
+    output.append(UNIX_LINE_SEPARATOR);
+    encoder.encode(input, output);
+    encoder.finalize(output);
+    output.flip();
+    return output.toString().trim()
+      .concat(UNIX_LINE_SEPARATOR).concat("-----END ").concat(encodeType.getType()).concat("-----");
+  }
+
+  /**
+   * Retrieves the subject distinguished name (DN) of the provided X.509 certificate.
+   *
+   * The subject DN represents the identity of the certificate holder and typically includes information
+   * such as the common name (CN), organizational unit (OU), organization (O), locality (L), state (ST),
+   * country (C), and other attributes.
+   *
+   * @param cert           The X.509 certificate from which to extract the subject DN.
+   * @param includeSpaces  A boolean flag indicating whether to include spaces in the subject DN string.
+   * @return The subject DN string of the X.509 certificate. If {@code includeSpaces} is true,
+   *         the DN string will be returned in its original format. If {@code includeSpaces} is false,
+   *         the DN string will be returned without spaces between attributes.
+   *
+   * @throws NullPointerException  If the provided certificate is null.
+   */
+  public static String subjectDN(final X509Certificate cert, final boolean includeSpaces)
```

Comment:
What is the use case for the `includeSpaces` flag?

Reply:
Per requirement: https://github.com/vt-middleware/cryptacular/issues/67
For (2), consider flags or overloaded variants that optionally produce output containing spaces

Reply:
In my experience this aspect of DN formatting can matter in some use cases, thus the requirement. While one could argue that "any implementation that depends on the formatting of the DN is broken," searching by exact DN is common and thus a counter-example to that argument. Perhaps unfortunately, whitespaces matter sometimes.

Reply:
Why not return an `X500Principal` and leave the String representation to the invoker?
If that doesn't seem acceptable, consider accepting a `format` String input, which could then be declared as:
`X500Principal#RFC2253` or `X500Principal#CANONICAL` to allow the invoker concrete control over the output.